PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,91 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class extends the LocalInspectionTool to check for the use of connection strings in the code.
+ * If the method is called from an Azure client class, a problem is registered with the suggestion message.
+ */
+public class ConnectionStringCheck extends LocalInspectionTool {
+
+    private static List <String> ruleConfigurations;
+    private static final Logger LOGGER = Logger.getLogger(ServiceBusReceiverAsyncClientCheck.class.getName());
+
+
+    // Load the rule configurations from the configuration file
+    static {
+        try {
+            ruleConfigurations = loadRuleConfigurations();
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "Error loading client data", e);
+        }
+    }
+
+    private static final String RULE_NAME = "ConnectionStringCheck";
+    private static final String METHOD_TO_CHECK_KEY = "method_to_check";
+    private static final String SUGGESTION_KEY = "antipattern_message";
```

Comment:
These can be moved to a common util class or the loadConfiguration can take these parameters to only get data relevant to this Rule.

Reply:
return a json model class with methods to retrieve relevant information

Reply:
done!